src/edit.js

```diff
@@ -25,26 +31,70 @@ import Block from './block';
  * @see https://developer.wordpress.org/block-editor/developers/block-api/block-edit-save/#edit
  * @return {WPElement} Element to render.
  */
-export default function Edit( { attributes, setAttributes } ) {
+function ProgressEdit( {
+	attributes,
+	setAttributes,
+	messageColor,
+	setMessageColor,
+	progressColor,
+	setProgressColor,
+} ) {
 	const { freeShippingFrom } = attributes;
 
 	return (
 		<div { ...useBlockProps() }>
 			<InspectorControls>
 				<PanelBody title="Settings" initialOpen={ true }>
 					<TextControl
-						label="Free shipping from"
-						help="Provide the value in your store currency."
+						label={ __(
+							'Free shipping from',
+							'free-shipping-progress-bar'
+						) }
+						help={ __(
+							'Provide the value in your store currency.',
+							'free-shipping-progress-bar'
+						) }
 						value={ freeShippingFrom }
 						onChange={ ( value ) =>
 							setAttributes( {
 								freeShippingFrom: value,
 							} )
 						}
 					/>
+					<PanelColorSettings
+						title={ __(
+							'Color Settings',
+							'free-shipping-progress-bar'
+						) }
+						colorSettings={ [
+							{
+								label: __(
+									'Message Color',
+									'free-shipping-progress-bar'
+								),
+								value: messageColor.color,
+								onChange: setMessageColor,
+							},
+							{
+								label: __(
+									'Progress Bar Color',
+									'free-shipping-progress-bar'
+								),
+								value: progressColor.color,
+								onChange: setProgressColor,
+							},
+						] }
+					></PanelColorSettings>
```

Comment:
This is correct, however, you're not consuming the values correctly.

Reply:
I do not see what's wrong here. Isn't this identical to https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/e23b51119dfa5c11e67ab0b3cbd89d49189a8569/assets/js/atomic/blocks/product-elements/title/edit.js#L90-L105?

Reply:
Yes, this code part is correct, I was asserting that no changes are needed here, you just need to change how you consume the colors above.

Reply:
I'm not sure which line you're referring to. Would you mind telling me the line number, so that can see what you mean?